src/rpc/blockchain.cpp

```diff
@@ -1916,6 +1923,435 @@ static UniValue savemempool(const JSONRPCRequest& request)
+                        if (script_type == OutputScriptType::UNKNOWN) throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid script type");
+
+                        // get internal and external scripts and add it to the containers
+                        CScript script = GetScriptForDestination(GetDestinationForKey(k_external.pubkey, script_type));
+                        assert(!script.empty());
+                        needles.insert(script);
+                        temp_keystore.AddWatchOnly(script);
+
+                        script = GetScriptForDestination(GetDestinationForKey(k_internal.pubkey, script_type));
+                        assert(!script.empty());
+                        needles.insert(script);
+                        temp_keystore.AddWatchOnly(script);
+                    }
+                }
+            }
+            else if (script_uni.isStr()) {
+                // type: script
+                // check and add the script to the scan containers (needles array, temp keystore)
+                std::vector<unsigned char> scriptData(ParseHexV(script_uni, "script"));
+                CScript script(scriptData.begin(), scriptData.end());
+                // TODO: check script: max length, has OP, is unspenable etc.
+                needles.insert(script);
+                temp_keystore.AddWatchOnly(script);
+            }
+        }
+
+        CMutableTransaction tx;
+        // Scan the unspent transaction output set for inputs
+        UniValue unspents(UniValue::VARR);
+        std::vector<CTxOut> input_txos;
+        std::map<COutPoint, Coin> coins;
+        g_should_abort_scan = false;
+        g_scan_progress = 0;
+        int64_t count = 0;
+
+        // flush utxo state and start the scan
+        FlushStateToDisk();
+        bool res = pcoinsdbview->FindScriptPubKey(g_scan_progress, g_should_abort_scan, count, needles, coins);
+
+        // report back
+        result.push_back(Pair("success", res ? "yes" : "no"));
+        result.push_back(Pair("searched_items", count));
+
+        int nIn = 0;
+        for (const auto& it : coins) {
+            const COutPoint& outpoint = it.first;
+            const Coin& coin = it.second;
+            const CTxOut& txo = coin.out;
+            tx.vin.emplace_back(outpoint.hash, outpoint.n);
+            tx.vin.back().nSequence = MAX_BIP125_RBF_SEQUENCE; //enforce BIP125
+
+            // Fill in dummy signatures for fee calculation, ignore signature verification.
+            const CScript& scriptPubKey = txo.scriptPubKey;
+            SignatureData sigdata;
+            ProduceSignature(temp_keystore, DUMMY_SIGNATURE_CREATOR, scriptPubKey, sigdata);
+            UpdateInput(tx.vin.at(nIn), sigdata);
+
+            input_txos.push_back(txo);
+            total_in += txo.nValue;
+
+            UniValue unspent(UniValue::VOBJ);
+            unspent.push_back(Pair("txid", outpoint.hash.GetHex()));
```

Comment:
Should use `pushKV` instead, `push_back(Pair(...))` occurrences were replaced recently.